spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkSchemaUtil.java

```diff
@@ -127,6 +137,76 @@ public static Schema convert(StructType sparkType) {
     return new Schema(converted.asNestedType().asStructType().fields());
   }
 
+  public static Schema convert(List<org.apache.spark.sql.connector.catalog.Column> columns) {
```

Comment:
Since this is the Utility which converts default values when doing the conversion to the Iceberg schema and `convert(StructType sparkType)` does not, we should probably deprecate the other one so that users don't use that API and potentially drop their default values